test/sagemaker_tests/pytorch/training/resources/mnist/smmodelparallel_pt_mnist.py

```diff
@@ -102,7 +102,10 @@ def train(args, model, scaler, device, train_loader, optimizer, epoch):
             scaler.step(optimizer)
             scaler.update()
         else:
-            optimizer.step()
+            # some optimizers like adadelta from PT 1.8 dont like it when optimizer.step is called with no param
+            # this is a bug in PT 1.8
+            if len(list(model.local_parameters())) > 0:
```

Comment:
What happens when you call the optimizer step with no params? Should this be written as `Version(framework_version) < Version(1.8) or len(list(model.local_parameters())) > 0`?

Reply:
Sure, but `optimizer.step()` is never called if `model.local_parameters()` is empty, right? Is this alright?

Reply:
Yeah, if there's no local params, then optimizer.step has nothing to do